tests/system/knowledge_base/test_knowledge_base.py

```diff
@@ -260,6 +287,13 @@ def test_upsert_happy_path(knowledge_base, documents, encoded_chunks):
     assert_chunks_in_index(knowledge_base, encoded_chunks)
 
 
+def test_upsert_with_security_scanning(knowledge_base_fw, malicious_documents):
```

Comment:
Hello, great work here, thanks for adding this, it's a great feature to have.

In this test, is it possible to ensure that a `AIFirewallError` is raised instead of a generic `ValueError`? I think that'd be more accurate (you could even ensure the message contains the relevant error text). 

Additionally, is it possible to add tests to account for the event of someone adding in a few malicious docs within a larger batch of good docs? I'd assume that's the more likely scenario IRL.

Reply:
@aulorbe many thanks for the prompt review! Really appreciated. Addressing your comments now!